include/vcpu/arch/intel_x64/vcpu.h

```diff
@@ -38,7 +40,28 @@ class vcpu : public bfvmm::intel_x64::vcpu
     ///
     vcpu(vcpuid::type id) :
         bfvmm::intel_x64::vcpu{id}
-    { }
+    {
+        enable_msr_trapping();
+
+        auto handler = this->exit_handler();
```

Comment:
Couple things:

- This should be in its own function similar to how we enable the other apis. 
- We should not pass the VMCS to the class as it can get it from the exit handler. 
- The code should look more like this to stay consistent
- Use msrs() instead of directly accessing the member. 

```
m_irqmgr = std::make_unique<eapis::intel_x64::irq_manager>(
    this->exit_handler(), 
    this->msrs()
);
```

Reply:
The vmcs is a private member of exit_handler